src/lib/pubkey/pk_algs.cpp

```diff
@@ -346,47 +325,82 @@ create_private_key(const std::string& alg_name,
 #if defined(BOTAN_HAS_RSA)
    if(alg_name == "RSA")
       {
-      const size_t rsa_bits = (params.empty() ? 3072 : to_u32bit(params));
-      return std::make_unique<RSA_PrivateKey>(rng, rsa_bits);
+      const size_t modulus_bits = [&]() -> size_t
+         {
+         if(params.empty())
+            return 3072;
+         return to_u32bit(params);
+         }();
+
+      return std::make_unique<RSA_PrivateKey>(rng, modulus_bits);
       }
 #endif
 
 #if defined(BOTAN_HAS_MCELIECE)
    if(alg_name == "McEliece")
       {
-      std::vector<std::string> mce_param =
-         split_on(params.empty() ? "2960,57" : params, ',');
-
-      if(mce_param.size() != 2)
-         throw Invalid_Argument("create_private_key bad McEliece parameters " + params);
-
-      size_t mce_n = to_u32bit(mce_param[0]);
-      size_t mce_t = to_u32bit(mce_param[1]);
-
-      return std::make_unique<McEliece_PrivateKey>(rng, mce_n, mce_t);
+      const auto [n, t] = [&]() -> std::pair<size_t, size_t>
+         {
+         if(params.empty())
+            return std::make_pair(2960, 57);
```

Comment:
Nit: I believe this would do the trick with less code, because the expected return type is stated in the lambda declaration above. Caveat: I didn't try it. 😨

```suggestion
            return {2960, 57};
```